app/scenes/load_team/load_team.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+
+import Loading from 'app/components/loading';
+
+import {RequestStatus} from 'service/constants';
+
+export default class LoadTeam extends React.Component {
+    static propTypes = {
+        teams: React.PropTypes.object.isRequired,
+        myMembers: React.PropTypes.object.isRequired,
+        teamsRequest: React.PropTypes.object.isRequired,
+        currentTeam: React.PropTypes.object,
+        actions: React.PropTypes.shape({
+            goToChannelView: React.PropTypes.func.isRequired,
+            handleTeamChange: React.PropTypes.func.isRequired,
+            initWebsocket: React.PropTypes.func.isRequired
+        }).isRequired
+    };
+
+    componentWillMount() {
+        this.props.actions.initWebsocket();
```

Comment:
Yes, it'll be another PR. I wanted to get the rest of this merged before it conflicts with everything else again.

I think we could actually move the websocket stuff to the ChannelView. Since it'll always be at the root of the stack now, we can have it watch for the app coming out of the background to fire off some stuff.